docs/contributing/documentation_guidelines.md

```diff
@@ -289,27 +475,63 @@ Images are hard to keep up-to-date as documentation evolves, but can be worthwhi
 
 </details>
 
-### Autodoc
+### Docstrings and API reference (autodoc)
 
-With its [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) feature, Sphinx supports writing documentation in Python docstrings for subsequent integration in the project’s documentation pages. This is a very powerful feature that we highly recommend using to document Wagtail’s APIs.
+With its [autodoc](inv:sphinx#ext-autodoc) feature, Sphinx supports writing documentation in Python docstrings for subsequent integration in the project’s documentation pages. This is a very powerful feature that we highly recommend using to document Wagtail’s APIs.
+
+Modules, classes, and functions can be documented with docstrings. Class and instance attributes can be documented with docstrings (with triple quotes `"""`) or doc comments (with hash-colon `#:`). Docstrings are preferred, as they have better integration with code editors. Docstrings in Python code must be written in reStructuredText syntax.
+
+```py
+SLUG_REGEX = r'^[-a-zA-Z0-9_]+$'
+"""Docstring for module-level variable ``SLUG_REGEX``."""
+
+class Foo:
+    """Docstring for class ``Foo``."""
+
+    bar = 1
+    """Docstring for class attribute ``Foo.bar``."""
+
+    #: Doc comment for class attribute ``Foo.baz``.
+    #: It can have multiple lines, and each line must start with ``#:``.
+    #: Note that it is written before the attribute.
+    #: While Sphinx supports this, it is not recommended.
+    baz = 2
```

Comment:
I didn't know you can write docstrings for attributes too, I thought you had to use the hash-colon syntax.

Apparently docstrings are better supported in code editors. See the following:

![image](https://github.com/user-attachments/assets/cbd444be-9253-45f2-9ff1-ecc7ec617759)


![image](https://github.com/user-attachments/assets/466dcf16-5109-455b-9731-782d71222b5b)

While I don't think we've adopted this pattern for attributes, I think we should use it moving forward. It's a big win for editor integration.

Reply:
hmm, nice.